confgen/api.go

```diff
@@ -0,0 +1,34 @@
+// Package confgen provides a mechanism by which a supplied configuration is converted into a
+// set of configuration files to be supplied to ProxyFS (proxyfsd), Samba (smbd et. al.), and
+// NFS (nfsd). In (at least) the case of Samba, multiple configuration files will be produced
+// with a per-IP path.
+//
+// Only the configuration files for the Cluster.WhoAmI-identified Peer will be produced though,
+// in the case of the ProxyFS configuration file, all VolumeGroups will be included given that
+// ProxyFS needs to know the mapping of VolumeGroup to Peer. The Cluster.WhoAmI can either already
+// be present in the `confFilePath` or be provided as an element of the `confOverrides` argument
+// in each of the APIs below.
+//
+// As would be typical in a deployment where the placement of VolumeGroups on Peers in dynamic,
+// the `confFilePath` would provide `VirtualIPAddr` values as opposed to fixed `PrimaryPeer`
+// assignments. It is the callers responsibility to compute the assignments of VolumeGroups
+// to Peers and either modify the provided `confFilePath` to include PrimaryPeer values or to
+// supply those via the `confOverrides` (e.g. "VolumeGroup:CommonVolumeGroup.PrimaryPeer=Peer0").
+package confgen
+
+// ComputeInitial takes a supplied ConfFile, overlays ConfOverrides, and computes an initial
+// set of configuration files that are used by a per-IPAddr set of Samba instances as well as
+// NFSd & ProxyFS.
```

Comment:
How does ss-proxyfs-monitord pass the mapping of VGs -> nodes?
Is that in the confOverrides array?
Wouldn't it be easier to pass this as a map?   How do you consume this?

Reply:
I think the comment at the top kind of gives an example, no?
You would include in the confOverrides []string a string like "VolumeGroup:whatever.PrimaryPeer=PeerWhateverGUID".

Reply:
As for it being a map, that's true for the API but not for the cmdline version. Not sure how better to serialize a "map" for the cmdline :-)

Reply:
I should also say that this same mechanism is how the Cluster.WhoAmI is set as well...

Reply:
ok